swiftly-solution/swiftly
# feat(translation/scripting): Added supported client language

# [Feature] Add support for fetching translations based on client language

_This pull request introduces a change to the API, allowing the `FetchTranslation` script to accept an optional `playerId` parameter. If `playerId` is provided, translations will be fetched based on the client's language. If the parameter is omitted, translations will default to the core language._

### Changes:
- Updated `FetchTranslation` to handle an optional `playerId` parameter.
- When `playerId` is provided, the translation is fetched based on the client's language.
- If `playerId` is not provided, the translation defaults to the core language.

### Steps to test:
1. Call `FetchTranslation` with and without the `playerId` parameter.
2. Ensure that translations are correctly fetched based on the provided client language or fall back to the core language when `playerId` is absent.
3. Scripting test plugin:

```lua
commands:Register("test", function (playerid, args, argc, silent, prefix)
    local player = GetPlayer(playerid)
    if not player or not player:IsValid() then return end
    player:SendMsg(MessageType.Chat, "This is default core translation: " .. FetchTranslation("core.forever"))
    player:SendMsg(MessageType.Chat, "This is client translation: " .. FetchTranslation("core.forever",playerid ))
end)
```

### Checklist:
- [x] Code works as expected locally
- [x] Code adheres to project standards
